### ISSUE TITLE
Accessibility findings for spot check dev review for 10-10CG Family Caregiver Application for Benefits

### ISSUE THREAD

**Issue by LinnJS** [department-of-veterans-affairs/va.gov-team#5854]
## Instructions

1. Navigate to https://vetsgov-pr-11352.herokuapp.com/caregivers

2. Should be redirected to a introduction page, that outlines the caregiver process and has additional info for definitions 

3. Start the form by clicking the `Start your application without signing in` button

4. Form should be complete with a required Veteran/Service Member section, as well as a required Primary Caregiver section

5. Should have two optional Secondary One and Secondary Two sections with additional info for definitions 

6. Once required sections are filled out, you should see a review page that requires each name to be typed into box and must match previously entered first and last name exactly (case sensitive).

7. Once reviewed as signed should submit and direct you to a confirmation page.

### For reference

**VSA Development Review** - [https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/accessibility/review-process.md#development-review](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsa/accessibility/review-process.md#development-review)

## Team Name

Caregiver Team

## Product/Feature Name

10-10CG Application for Family Caregiver Benefits 

## Product Description

We're creating a new online application for the Program of Comprehensive Assistance for Family Caregivers (PCAFC). The application needs to be filled out by the Veteran, the primary caregiver, and any secondary caregivers that are applying.

## Checkpoint Stage

[ ] **Design-Development Intent Checkpoint** (Review design to discuss approach to take with development, identify what components and semantic HTML would be appropriate, identify any accessibility issues and how to address) - This would happen as designs are passed to development, before development begins.

[X] **Spot Check Review Checkpoint** (Review and follow the steps in the [508 Checklist](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/508-checklist.md) first.) This happens well before launch, so that the 508 office can review and any issues addressed *before* launch.

[ ] **Pre-Launch 508 Review** Once your code is stable in Staging (Review [Pre-Launch 508 Review platform-level documentation](https://github.com/department-of-veterans-affairs/va.gov-team/edit/master/platform/accessibility/508-request-prelaunch-review.md)) **All applications and features on the Veteran-facing Services Platform are required to be accessible and 508 compliant.** This includes adherence to the updated [Section 508](https://www.section508.gov/) standards and the [Web Content Accessibility Guidelines (WCAG)](https://www.w3.org/TR/WCAG20/) Level A and AA success criteria.

## Point of contact

@LinnJS (Justin Linn) @ala-yna 

## Staging URL and Credentials

https://vetsgov-pr-11352.herokuapp.com/caregivers
## Next steps

1. Accessibility specialist will create an epic in Zenhub to capture issues.

2. Team will coordinate another review after work is updated by @-comment in Zenhub in the Github issue.

3. Once all issues are addressed, the ticket is closed with a final comment, “All issues are addressed”, @-comment developer who opened issue, and close ticket/issue.

**Comment by 1Copenut** [comment]
@LinnJS @jenstrickland will be your accessibility point person on this review. I'll remove my name from the Assignees, but will be available to assist her or consult as needed.

**Comment by jenstrickland** [comment]
@LinnJS I'll be out of office Feb 14 and 17, and will work on this on Feb 18. cc @ala-yna

**Comment by jenstrickland** [comment]
Since this is a casual spot check, rather than creating issue tickets for each concern, here's a bulleted list of items to address, and each page in a separate comment. 

Framework for feedback:
```diff
- "Must" for if the feedback must be applied 
! "Should" if the feedback is best practice 
+ "Consider" for suggestions/enhancements 
```

**Overall**
- Must: Do you have the axe and axe-coconut tests in your build process? | Reference: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/508-accessibility-best-practices.md#automated-accessibility508-testing
<hr>

**Comment by jenstrickland** [comment]
**On the [Introduction page](https://vetsgov-pr-11352.herokuapp.com/caregivers/introduction)** 

- Must: The breadcrumbs component is missing above the H1, resulting in the H1 being too close to the main navigation.


- Must: The Start your Application button must look like a button. | Reference: https://design.va.gov/components/buttons ![Caregiver-StartYourApp-button](https://user-images.githubusercontent.com/57469/75492531-f9853300-5985-11ea-816f-58d6c038c0f4.png)


- Must: `<ul>` and `<ol>` must only directly contain `<li>`, `<script> `or `<template>` elements - Within the steps of the form, the last item is currently a `<div>` but MUST be an `<li>`. | Reference: https://dequeuniversity.com/rules/axe/3.4/list?application=AxeChrome
![Caregiver-div-instead-of-li](https://user-images.githubusercontent.com/57469/75494786-00627480-598b-11ea-95ea-8e2bde08113f.png)

- Must: Heading levels should only increase by one - The current headings go from an `<h1>` to an `<h4>`, but should proceed semantically as `<h1> <h2> <h3>, etc`. If you want an `<h2>` to look like an `<h4>` there are utility classes for that. | Reference: https://design.va.gov/design/typography#headings / https://dequeuniversity.com/rules/axe/3.4/heading-order?application=AxeChrome
```html
<h4>Can I get benefits through the Program of Comprehensive Assistance for Family Caregivers?</h4>

```

- Must: Interactive elements must be spaced sufficiently to prevent errors - Generally, place 1.5rem between interactive items.
![Caregiver-spacing-of-interactive-elements](https://user-images.githubusercontent.com/57469/75496266-b085ac80-598e-11ea-813d-2bd67754ae73.png)


> The Heroku link returned a 404 at 6:30pm, so I'll continue tomorrow.

**Comment by jenstrickland** [comment]
### 1 of 3 Veteran/Service Member

- Same as introduction, regarding breadcrumbs component

- Must: Heading levels should only increase by one - Same as on introduction page, use utility classes to maintain semantic heading order and visual design
```html
<h4><span class="form-process-step current">1</span> <span class="form-process-total">of 3</span> VETERAN/SERVICE MEMBER</h4>
```

### 2 of 3 PRIMARY FAMILY CAREGIVER

- Must: For links to learn more about Primary and Secondary Caregiver, same as on the introduction page, provide sufficient space to minimize errors (~1.5rem).


### 3 of 3 Review Application

- Must: Same as elsewhere, Heading levels should only increase by one - `<h4 tabindex="-1"><span class="form-process-step current">3</span> <span class="form-process-total">of 3</span> Review Application</h4>` should be an `<h2>` with utility class to look like an `<h4>`. The `<h5>Need Help?</h5>` needs the same technique.

- Must: Form elements must have labels - On form elements, a the `for` attribute of a `label` must match the `id` of the `input`. This is an issue on both of the full name inputs.
![Caregiver-labels-inputs](https://user-images.githubusercontent.com/57469/75497329-ed06d780-5991-11ea-9052-83742767153e.png)

**Comment by jenstrickland** [comment]
**Accessibility spot check is complete.** 

There are design and content revisions, as discussed with @ala-yna - this is early yet in the effort.
